### PR TITLE
feat: search

### PR DESCRIPTION
**Pull request by TheodoreChu**
## Summary

This PR adds the scripts and styles necessary for search to `index.html` and adds the appropriate keybindings to the CodeMirror configuration in `main.js`. 

The official CodeMirror demo for search is available here: https://codemirror.net/demo/search.html

Keybindings:
> Demonstration of primitive search/replace functionality. The keybindings (which can be configured with custom keymaps) are:
Ctrl-F / Cmd-F Start searching
Ctrl-G / Cmd-G Find next
Shift-Ctrl-G / Shift-Cmd-G Find previous
Shift-Ctrl-F / Cmd-Option-F Replace
Shift-Ctrl-R / Shift-Cmd-Option-F Replace all
Alt-F Persistent search (dialog doesn't autoclose, enter to find next, Shift-Enter to find previous)
Alt-G Jump to line

## Screenshots

<kbd>Ctrl</kbd> + <kbd>F</kbd>: Search once
image
After pressing <kbd>Enter</kbd>
image

<kbd>Alt</kbd> + <kbd>F</kbd>: Persistent search
image

## Potential Issues
- The scrollbar is not visible as of `v1.3.6` (prior to this PR). If the scrollbar was visible, then search results would appear as highlights on the scrollbar.
- The first line is not entirely visible when using search. This is also an issue with the official demo. We can fix this by adding a little bit more padding to the top of the `.CodeMirror` class, but fixing this may not be necessary. I suppose it's a matter of preference.

### CODE REVIEW

**Review by johnny243 — CHANGES_REQUESTED** [review]
An error is thrown when using the search functionality:

image

Add `annotatescrollbar.js` script in the `head` element to solve this:

```html
<script src="vendor/codemirror/addon/scroll/annotatescrollbar.js"></script>
```
---
> The scrollbar is not visible as of v1.3.6 (prior to this PR). If the scrollbar was visible, then search results would appear as highlights on the scrollbar.

Removing `main.css` from the `index.html` file shows both scrollbars, so the problem might be there...

**Comment by TheodoreChu** [comment]
> Removing `main.css` from the `index.html` file shows both scrollbars, so the problem might be there...

Yes, I think the issue is that the padding around `.CodeMirror` makes it larger than the `#wrapper` container, which has horizontal-overflow hidden. So, I removed the left, right, and bottom padding. Then I adjusted the vertical padding so the search dialog does not overlap with the first line. Do you think that's a good idea?

This is how I found that the scroll bar existed but was hidden from sight:
image

This is how the editor looks in-app with search closed:
image

This is how the editor looks in-app with search open:
image

File: src/main.scss [hunk 1]
```diff
@@ -58,8 +58,9 @@ body, html {
   height: 100%;
   resize: none;
 
-  padding: 15px;
-  padding: 8px;
+  // Height of the search dialog. 
+  // If it's too big, subtract 4px (padding-top of .CodeMirror-lines) or remove entirely
+  padding-top: 21.08px;
```
Comment (johnny243) [inline src/main.scss:63]: I think we should be good without this line

Comment (TheodoreChu) [reply src/main.scss:63]: So it's OK / preferred to have no padding on top?

Comment (johnny243) [reply src/main.scss:63]: Yup, having an empty space on top when editing feels like something should be there. The demo doesn't have it, so it's fine

Comment (TheodoreChu) [reply src/main.scss]: OK, done in https://github.com/standardnotes/code-editor/pull/25/commits/314be5e81eb4a6c00c3b69b45b4d2eb9defbbda9

**Review by johnny243 — APPROVED** [review]

**Pull request merged**